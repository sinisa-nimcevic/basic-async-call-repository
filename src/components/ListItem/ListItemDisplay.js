import React from 'react';
import classname from 'classname';
import './ListItem.scss';

const ListItemDisplay = ({ item, handleClick }) => {
  const { isActive, id, title } = item;

  return (
    <li onClick={handleClick} className={classname('list-item', isActive && 'list-item--active')}>
      <span className="list-item__id">{id}</span>
      <span className="list-item__content">{title}</span>
    </li>
  );
};

export default ListItemDisplay;
