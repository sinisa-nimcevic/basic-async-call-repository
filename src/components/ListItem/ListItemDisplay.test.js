import { fireEvent, render } from '@testing-library/react';
import { toMatchDiffSnapshot } from 'snapshot-diff';
import ListItemDisplay from './ListItemDisplay';

expect.extend({ toMatchDiffSnapshot });

const dummyItem = {
  id: 1,
  title: 'Dummy item',
  isActive: false,
};

const dummyItemActive = {
  ...dummyItem,
  isActive: true,
};

const dummyHandleClick = jest.fn();

describe('GIVEN ListItemDisplay', () => {
  describe('WHEN is rendered', () => {
    it('THEN displays item id and title', () => {
      const { getByText } = render(
        <ListItemDisplay item={dummyItem} handleClick={dummyHandleClick} />
      );

      expect(getByText(dummyItem.id)).toBeInTheDocument();
      expect(getByText(dummyItem.title)).toBeInTheDocument();
    });
  });

  describe('WHEN isActive is false', () => {
    it('THEN has different snapshot to isActive true', () => {
      const a = render(<ListItemDisplay item={dummyItem} handleClick={dummyHandleClick} />);
      const b = render(<ListItemDisplay item={dummyItemActive} handleClick={dummyHandleClick} />);

      expect(a).toMatchDiffSnapshot(b);
    });
  });

  describe('WHEN item is clicked', () => {
    it('THEN calls handleClick once', () => {
      const { getByText } = render(
        <ListItemDisplay item={dummyItem} handleClick={dummyHandleClick} />
      );
      fireEvent.click(getByText(dummyItem.title));
      expect(dummyHandleClick).toHaveBeenCalledTimes(1);
    });
  });
});
