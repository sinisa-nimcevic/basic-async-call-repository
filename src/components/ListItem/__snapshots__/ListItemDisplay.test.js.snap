// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`GIVEN ListItemDisplay WHEN isActive is false THEN has different snapshot to isActive true 1`] = `
"Snapshot Diff:
- First value
+ Second value

@@ -36,11 +36,11 @@
        </li>
      </div>
    </body>,
    \\"container\\": <div>
      <li
-       class=\\"list-item\\"
+       class=\\"list-item list-item--active\\"
        data-testid=\\"list-item-display-1\\"
      >
        <span
          class=\\"list-item__id\\"
        >"
`;
